### PR TITLE
Triplet Loss

### PR DESCRIPTION
**Pull request by emircanerol**
## Proposed changes

Added triplet loss, test for the loss and appended it to docs.src.nn.losses for documentation.

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: python/mlx/nn/losses.py [hunk 1]
```diff
@@ -232,6 +232,41 @@ def smooth_l1_loss(
     return _reduce(loss, reduction)
 
 
+def triplet_loss(
+    anchors: mx.array,
+    positives: mx.array,
+    negatives: mx.array,
+    axis: int = -1,
+    alpha: float = 1.0,
+    eps: float = 10e-6,
```
Comment (awni) [inline python/mlx/nn/losses.py:241]: Should this be `1e-6`?

File: docs/src/python/nn/losses.rst [hunk 2]
```diff
@@ -16,3 +16,4 @@ Loss Functions
    mse_loss
    nll_loss
    smooth_l1_loss
+   triplet_loss
```
Comment (awni) [inline docs/src/python/nn/losses.rst:19]: 👍

**Review by awni — COMMENTED** [review]
Looks really nice! Thanks for adding this! I left a few minor comments then we can merge it.

**Review by awni — CHANGES_REQUESTED** [review]
Meant to request changes.

**Comment by emircanerol** [comment]
Thank you for your feedback. I checked my last commit with `pre-commit run --all-files`, and compared the results with PyTorch triplet margin loss. Looks like there are no problems. Can you re-review the PR, please?

File: python/mlx/nn/losses.py [hunk 3]
```diff
@@ -232,6 +232,47 @@ def smooth_l1_loss(
...
+    margin: float = 1.0,
+    eps: float = 1e-6,
+    reduction: str = "none",
+) -> mx.array:
+    r"""
+    Computes the triplet loss for a set of anchor, positive, and negative samples.
+
+    .. math::
+
+       L_{\text{triplet}} = \max\left(\|A - P\|_p - \|A - N\|_p + \margin, 0\right)
```
Comment (awni) [inline python/mlx/nn/losses.py:250]: \margin is not defined. Maybe use `\alpha` and just make a note below

```
where :math:`\alpha` is the ``margin``.
```

**Review by awni — CHANGES_REQUESTED** [review]
One more thing to fix in the tex :)

**Comment by emircanerol** [comment]
Thank you for your support with my first open source contribution :)

**Review by awni — APPROVED** [review]
🚀

**Pull request merged**